### PR TITLE
Added `distributed.utils.key_split` functionality to `dask.utils.key_split`

### PR DESCRIPTION
**Pull request by lukeconibear**
- [x] Helps close [`distributed` issue 6313](https://github.com/dask/distributed/issues/6313)
  - Companion PR from `distributed` is [Replaced `distributed.utils.key_split` with `dask.utils.key_split` #7005](https://github.com/dask/distributed/pull/7005)
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by ncclementi** [comment]
add to allow list

**Comment by ncclementi** [comment]
@fjetter can we hand over this to a distributed maintainer/contributor for review? Since it touches both repos.

File: dask/utils.py [hunk 1]
```diff
@@ -1837,7 +1840,7 @@ def key_split(s):
     try:
         words = s.split("-")
         if not words[0][0].isalpha():
-            result = words[0].strip("_'()\"")
+            result = words[0].split(",")[0].strip("_'()\"")
```
Comment (fjetter) [inline dask/utils.py:1843]: This is very interesting. Can you add a test for this?

Comment (fjetter) [reply dask/utils.py:1843]: I'm sure this is fine, I'm just surprised to see these implementations to drift. FWIW this change was introduced in https://github.com/dask/distributed/pull/1566

Comment (fjetter) [reply dask/utils.py:1843]: Ok, just saw that we're executing doc tests.

**Review by fjetter — APPROVED** [review]
Thank you @lukeconibear !

**Comment by lukeconibear** [comment]
Thanks a lot!

**Pull request merged**